src/pkg/oci/common.go

```diff
@@ -75,38 +75,34 @@ func WithInsecureSkipVerify(insecure bool) Modifier {
 	}
 }
 
-// WithTargetPlatform sets the target platform for the remote
-func WithTargetPlatform(platform *ocispec.Platform) Modifier {
-	return func(o *OrasRemote) {
-		o.targetPlatform = platform
-	}
-}
-
 // WithSkeletonArch sets the target architecture for the remote to skeleton
-func WithSkeletonArch() Modifier {
-	return WithTargetPlatform(&ocispec.Platform{
+func WithSkeletonArch() ocispec.Platform {
+	return ocispec.Platform{
 		OS:           MultiOS,
 		Architecture: SkeletonArch,
-	})
+	}
 }
 
 // WithArch sets the target architecture for the remote
-func WithArch(arch string) Modifier {
-	return WithTargetPlatform(&ocispec.Platform{
+func WithArch(arch string) ocispec.Platform {
```

Comment:
```suggestion
func PlatformForArch(arch string) ocispec.Platform {
```
This function naming would help this be distinct from other uses of `With` in this file.

Reply:
updated